samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+   {
+       uint32_t i;
+       ret_code_t err_code;
+       uint32_t hash_test_vector_count = TEST_VECTOR_COUNT;
+
+       for (i = 0; i < hash_test_vector_count; i++)
+       {
+           test_vector_hash_t * p_test_vector = TEST_VECTOR_GET(i);
+
+           // TODO: Add test code here.
+
+           // Will goto exit_test_vector if fails.
+           TEST_VECTOR_ASSERT(err_code);
+
+           p_test_info->tests_passed++;
+
+   exit_test_vector: ;
+       }
+       return NRF_SUCCESS;
+   }
+
+Registering a test case
+-----------------------
+
+A new test case must be registered to the test_case_data section using lib_section_vars.
+The NRF_SECTION_ITEM_REGISTER macro below places the variable test_hash in a section named "test_case_data", which is initialized by main::
+
+   NRF_SECTION_ITEM_REGISTER(test_case_data, test_case_t test_hash) =
+   {
+       .p_test_case_name = "Hash test",        /**< Pointer to hash test case name. */
+       .setup = setup_test_case_hash,          /**< Pointer to hash test case setup function. */
+       .exec = exec_test_case_hash,            /**< Pointer to hash test case execute function. */
+       .teardown = teardown_test_case_hash     /**< Pointer to hash test case teardown function. */
+   };
+
+.. note::
+   The macro call to NRF_SECTION_ITEM_REGISTER must be done in a .c file.
+
+Test vectors
+------------
+
+A test vector is a set of inputs and expected outputs to verify the functionality provided in a test case.
+The ``p_test_vector_name`` variable is the only required member in the test_vector structures::
+
+   typedef const struct
+   {
+       const nrf_crypto_hash_info_t    * p_hash_info;                  /**< Pointer to hash info type. */
+       const ret_code_t                  expected_err_code;            /**< Expected error code from hash operation. */
+       const uint8_t                     expected_result;              /**< Expected result of hash operation. */
+       const hash_mem_mode_t             mode;                         /**< Hash memory operation. */
+       const uint32_t                    chunk_length;                 /**< Size of input chunks to hash function in bytes. */
+       const uint32_t                    update_iterations;            /**< Number of update iterations of input. */
+       const char                      * p_test_vector_name;           /**< Pointer to hash test vector name. */
+       const char                      * p_input;                      /**< Pointer to input message in hex string format. */
+       const char                      * p_expected_output;            /**< Pointer to expected message digest in hex string format. */
+   }test_vector_hash_t;
+
+Registering a test vector
+-------------------------
+
+Test vectors are added by registering them for a section defined in the test case code.
+The test vector is registered in the section ``test_vector_hash_data`` which is defined in the test case example ``exec_test_case_hash``.
```

Comment:
change "test_vector_hash_data" to "test_vector_hmac_data" and "exec_test_case_hash" to "exec_test_case_hmac"

Reply:
done